tools/preflight/helper.go

```diff
@@ -0,0 +1,448 @@
+							"cpu":    resource.MustParse("250m"),
+						},
+						Limits: map[corev1.ResourceName]resource.Quantity{
+							"memory": resource.MustParse("128Mi"),
+							"cpu":    resource.MustParse("500m"),
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "source-data",
+							MountPath: "/demo/data",
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "source-data",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return pod
+}
+
+// createVolumeSnapsotSpec creates pvc for volume snapshot
+func createVolumeSnapsotSpec(name, namespace, snapVer, pvcName string) *unstructured.Unstructured {
+	volSnap := &unstructured.Unstructured{}
+	volSnap.Object = map[string]interface{}{
+		"spec": map[string]interface{}{
+			"volumeSnapshotClassName": storageVolSnapClass,
+			"source": map[string]string{
+				"persistentVolumeClaimName": pvcName,
+			},
+		},
+	}
+	volSnap.SetName(name)
+	volSnap.SetNamespace(namespace)
+	volSnap.SetKind(internal.VolumeSnapshotKind)
+	volSnap.SetAPIVersion("snapshot.storage.k8s.io/" + snapVer)
+	volSnap.SetLabels(map[string]string{
+		labelTrilioKey:       labelTvkPreflightValue,
+		labelPreflightRunKey: resNameSuffix,
+		labelK8sPartOf:       labelK8sPartOfValue,
+	})
+
+	return volSnap
+}
+
+// createRestorePVCSpec creates pvc for restore (unused pvc as well)
+func createRestorePVCSpec(pvcName, dsName, storageClass, namespace string) *corev1.PersistentVolumeClaim {
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			StorageClassName: &storageClass,
+			Resources: corev1.ResourceRequirements{
+				Requests: map[corev1.ResourceName]resource.Quantity{
+					"storage": resource.MustParse("1Gi"),
+				},
+			},
+			DataSource: &corev1.TypedLocalObjectReference{
+				Kind:     internal.VolumeSnapshotKind,
+				Name:     dsName,
+				APIGroup: func() *string { str := "snapshot.storage.k8s.io"; return &str }(),
+			},
+		},
+	}
+
+	return pvc
+}
+
+// createRestorePodSpec creates a restore pod
+func createRestorePodSpec(podName, pvcName, namespace string) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PodSpec{
```

Comment:
add SA, imagepullsecret, etc to all pods

Reply:
Done